### PR TITLE
add support for 'none' BorderSizeValue to hide border

### LINKED ISSUE #4833: BorderSizeValue should support 'none' to hide the border
BorderSizeValue should support `none` such that one can conditional remove the border based on responsive value.

```typescript
export type BorderSizeValue =
  | 'thin'
  | 'thick'
  | 'thicker'
  | 'thickest';
```

```typescript
<View borderEndWidth={{ base: 'none', M: 'thin' }}>...</View>
```

### PR DESCRIPTION
**Pull request by pr7prashant**
Closes #4833 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

1. Using `none` as border size should display no border.
2. When using responsive break point, both using `none` and omitting the break point should display no border.
```
// Both should display no border
<View borderEndWidth={{ base: 'none', M: 'thin' }}>...</View>
<View borderEndWidth={{M: 'thin' }}>...</View>
```

### CODE REVIEW

File: packages/@react-spectrum/utils/src/styleProps.ts [hunk 1]
```diff
@@ -166,7 +166,9 @@ function borderColorValue(value: BorderColorValue, version = 5) {
 }
 
 function borderSizeValue(value: BorderSizeValue) {
-  return `var(--spectrum-alias-border-size-${value})`;
+  return value && value !== 'none'
+    ? `var(--spectrum-alias-border-size-${value})`
+    : undefined;
```
Comment (snowystinger) [inline packages/@react-spectrum/utils/src/styleProps.ts:171]: this won't work, border size defaults to `medium`
https://jsfiddle.net/qv5kp0zr/1/

We'll need to set the returned value to 0 instead

Comment (pr7prashant) [reply packages/@react-spectrum/utils/src/styleProps.ts]: Good catch. While testing it was working for me on storybook but it is better to return `0`. I've updated the PR. Thanks

**Comment by devongovett** [comment]
Do we need `'none'`? Isn't that equivalent to `null` or `undefined`?

**Comment by pr7prashant** [comment]
> Do we need `'none'`? Isn't that equivalent to `null` or `undefined`?

I added none so that the users can choose to omit or explicitly specify `none` . Please let me know if adding none is not required. I am okay with either approach :)

**Comment by LFDanLu** [comment]
@pr7prashant apologies about the PR highjacking, we were looking to get this in for an upcoming testing session. Still discussing amongst the team about what exact api we want here.

**Review by snowystinger — APPROVED** [review]

**Comment by pr7prashant** [comment]
> @pr7prashant apologies about the PR highjacking, we were looking to get this in for an upcoming testing session. Still discussing amongst the team about what exact api we want here.
> 
> EDIT: we've gone back to accepting "none" as an option, I've added some tests to cover the changes. Thanks for the contribution!

@LFDanLu No worries and thanks for the tests :)

**Review by devongovett — APPROVED** [review]

**Pull request merged**